internal/testutils/endpoint.go

```diff
@@ -45,7 +45,7 @@ func (b byAllFields) Less(i, j int) bool {
 // considers example.org. and example.org DNSName/Target as different endpoints
 func SameEndpoint(a, b *endpoint.Endpoint) bool {
 	return a.DNSName == b.DNSName && a.Target == b.Target && a.RecordType == b.RecordType &&
-		a.Labels[endpoint.OwnerLabelKey] == b.Labels[endpoint.OwnerLabelKey]
+		a.Labels[endpoint.OwnerLabelKey] == b.Labels[endpoint.OwnerLabelKey] && a.RecordTTL == b.RecordTTL
```

Comment:
I would bet that this should not be changed.
What would you expect to happen if only the TTL would be changed?
I would expect that it would create a new record if the TTL is the only thing which was changed.